### ISSUE TITLE
Incorrect handling of symlinks when creating file links in the cargo output

### ISSUE THREAD

**Issue by voidc** [intellij-rust/intellij-rust#3836]
## Environment

* **Intellij-Rust plugin version:** 0.2.97.2123-191
* **Rust toolchain version:** rustc 1.34.1 (fc50f328b 2019-04-24), cargo 1.34.0 (6789d8a0a 2019-04-01)
* **IDE name and version:** CLion 2019.1.3
* **Operating system:** Linux 5.0.13-arch1-1-ARCH

## Problem description

My project is located in the the directory `~/xyz/project`. The `~/xyz` directory is a symlink to `/mnt/xyz`. Whenever I click on a file link (e.g. `src/main.rs`) in cargo's build output, CLion opens `/mnt/xyz/project/src/main.rs` instead of `~/xyz/project/src/main.rs`. CLion does not recognize that these two files are the same and will not let me edit the opened file as it does not belong to the project. I would expect, that the file link opens `~/xyz/project/src/main.rs` which belongs to the project.

I'm not sure if this is an issue of this plugin or of the IntelliJ platform itself.
Potentially relevant file: https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/cargo/runconfig/filters/RegexpFileLinkFilter.kt

## Steps to reproduce

Recreate the environment described above and start a cargo build which fails.

**Comment by Andlon** [comment]
Just chiming in to say that I have the exact same issue.

**Comment by BuggStream** [comment]
Yes I have the same issue. In my case I am using fedora silverblue which has the `/home/<user>` symlinked to `/var/home/<user>`. In this case intellij rust does not recognize files in my `~/projects/<project>` directory as the same in the `var one`. Which means that auto completion is broken.

**Comment by Igosuki** [comment]
This issue resurfaces in latest EAP, symlinked file opened from the terminal are opened with the full path instead of the symlinked project path.
